api/v1/_email/PKPEmailHandler.inc.php

```diff
@@ -0,0 +1,173 @@
+				[
+					'pattern' => $this->getEndpointPattern() . '/{queueId}',
+					'handler' => [$this, 'process'],
+					'roles' => [ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER],
+				],
+			],
+		];
+		parent::__construct();
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize
+	 */
+	public function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.PolicySet');
+		$rolePolicy = new PolicySet(COMBINING_PERMIT_OVERRIDES);
+
+		import('lib.pkp.classes.security.authorization.RoleBasedHandlerOperationPolicy');
+		foreach ($roleAssignments as $role => $operations) {
+			$rolePolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, $role, $operations));
+		}
+		$this->addPolicy($rolePolicy);
+
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Create a jobs queue to send a bulk email to users in one or
+	 * more user groups
+	 *
+	 * @param ServerRequestInterface $slimRequest
+	 * @param APIResponse $response
+	 * @param array $args arguments
+	 * @return APIResponse
+	 */
+	public function create(ServerRequestInterface $slimRequest, APIResponse $response, array $args) {
+		$contextId = $this->getRequest()->getContext()->getId();
+
+		$requestParams = $slimRequest->getParsedBody();
+
+		$params = [];
+		foreach ($requestParams as $param => $val) {
+			switch ($param) {
+				case 'userGroupIds':
+					if (!is_array($val)) {
+						$val = explode(',', $val);
+					}
+					$params[$param] = array_map('intval', $val);
+			}
+		}
+
+		if (empty($params['userGroupIds'])) {
+			return $response->withJsonError('api.emails.400.missingUserGroups');
+		}
+
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+		foreach ($params['userGroupIds'] as $userGroupId) {
+			if (!$userGroupDao->contextHasGroup($contextId, $userGroupId)) {
+				return $response->withJsonError('api.emails.403.notAllowedUserGroup');
+			}
+		}
+
+		// Only permit emails to be sent to active users in this context
+		$params['status'] = 'active';
+		$params['contextId'] = $contextId;
+
+		$userIds = Services::get('user')->getIds($params);
+		$queueId = 'email_' . uniqid();
+		$batches = array_chunk($userIds, self::EMAILS_PER_JOB);
+		foreach ($batches as $userIds) {
+			Queue::push(function() use ($userIds, $contextId) {
+				$users = Services::get('user')->getMany([
+					'contextId' => $contextId,
+					'userIds' => $userIds,
+				]);
+				foreach ($users as $user) {
+					error_log('Email user id: ' . $user->getId());
+				}
+		}, [], $queueId, 'persistent');
+		}
+		return $response->withJson([
+			'queueId' => $queueId,
+			'totalJobs' => count($batches),
+		], 200);
+	}
+
+	/**
+	 * Process a jobs queue for sending a bulk email
+	 *
+	 * @param ServerRequestInterface $slimRequest
+	 * @param APIResponse $response
+	 * @param array $args arguments
+	 * @return APIResponse
+	 */
+	public function process(ServerRequestInterface $slimRequest, APIResponse $response, array $args) {
+		$countRunning = Capsule::table('jobs')
```

Comment:
Hmm, too bad the `DatabaseQueue` implementation in Laravel Queue doesn't provide access to this kind of information.